src/partition_mesh.py

```diff
@@ -179,7 +180,22 @@ def partition_metis(mesh: Mesh, numparts: int):
         cell = mesh.cells[i]
         cellData += list(cell)
         cellPtr.append(cellPtr[-1] + len(cell))
-    libmetis = cdll.LoadLibrary(os.path.abspath("libmetisAPI.so"))
+    binpath = os.path.dirname(__file__)
+    libpath = os.path.normpath(os.path.join(binpath, "../lib"))
+    os_type = platform.system()
+    if os_type == "Linux":
+        ext = ".so"
+    elif os_type == "Darwin":  # MacOS
+        ext = ".dylib"
+    else:
+        raise Exception("Unknown OS type")
+    if os.path.isfile(os.path.join(binpath, "libmetisAPI"+ext)):
+        libmetispath = os.path.join(binpath, "libmetisAPI"+ext)
+    elif os.path.isfile(os.path.join(libpath, "libmetisAPI"+ext)):
+        libmetispath = os.path.join(libpath, "libmetisAPI"+ext)
+    else:
+        raise Exception("libmetisAPI"+ext+" cannot found!")
```

Comment:
Is this really required, now that we have a proper installation in the `lib` directory?

Reply:
It is not required now, one can install and add the install path to required env variables. 
Thanks for reminder.

Reply:
Wait, let's keep something so that we can runn it from the build directory.

Reply:
Ready to merge.